spec/features/hiring_staff_can_edit_a_published_vacancy_spec.rb

```diff
@@ -66,6 +76,29 @@
         expect(page).to have_content('Assistant Head Teacher')
       end
 
+      scenario 'can edit job role for a legacy vacancy' do
+        # rubocop:disable Rails/SkipsModelValidations
+        vacancy.update_columns(job_role: [])
+        # rubocop:enable Rails/SkipsModelValidations
+
+        visit edit_school_job_path(vacancy.id)
+
+        expect(page).to have_content(I18n.t('jobs.job_details'))
+        expect(page).to have_content(I18n.t('messages.jobs.new_sections.message'))
+        expect(page.find('h2', text: I18n.t('jobs.job_details'))
+          .text).to include(I18n.t('jobs.notification_labels.new'))
+
+        click_link_in_container_with_text(I18n.t('jobs.job_role'))
+
+        fill_in_job_specification_form_fields(vacancy)
+
+        click_on 'Update job'
+
+        expect(page).to have_content(I18n.t('jobs.job_details'))
+        expect(page.find('h2', text: I18n.t('jobs.job_details'))
+          .text).to_not include(I18n.t('jobs.notification_labels.new'))
```

Comment:
Maybe we should also say:

```ruby
        expect(page).not_to have_content(I18n.t('messages.jobs.new_sections.message'))
```

to verify that the new_sections message disappears?